### PR TITLE
src: add SyntaxError

### PR DESCRIPTION
**Pull request by KevinEady**
Adds wrapper for `Napi::SyntaxError` from release of Node-API v9.

### CODE REVIEW

**Comment by KevinEady** [comment]
|Node Version|Link|Status|
|-|-|-|
|20|https://ci.nodejs.org/job/node-test-node-addon-api-new/7558/||
|21|https://ci.nodejs.org/job/node-test-node-addon-api-new/7559/||

File: doc/syntax_error.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# SyntaxError
+
```
Comment (mhdawson) [inline doc/syntax_error.md:2]: Should it say here that it's only supported in node-api version 9 and higher?

Comment (mhdawson) [reply doc/syntax_error.md:2]: That might be a gap though ?  I go to try to use an API and it's not there?  Either way agree that we don't need to be block this PR on it, but might be worth thinking about wether adding that info makes sense.

**Review by legendecas — APPROVED** [review]
LGTM % one nit.

**Comment by KevinEady** [comment]
Added `do not land` label until release of v7.0.0 is complete.

**Pull request merged**